[Wayne Tee] iP

waynetee opened the pull request:
# DukePro

- Its amazing

1. Truly amazing!

```java
return System.out.println("the best!");
```

- [x] love it 😋 

> wonderful!

[https://github.com/waynetee/ip](https://github.com/waynetee/ip)

`awesome`

_wooo**ho~~o~~o**000_

Commit:
Add Gradle support

waynetee committed:
Add greeting

waynetee committed:
Add echo and exit

waynetee committed:
Reformat code

waynetee committed:
Add add and list

waynetee committed:
Add Task class

waynetee committed:
Refactor list

waynetee committed:
Add done

waynetee committed:
Add todo

waynetee committed:
Add deadline and events

waynetee committed:
Add test

waynetee committed:
Add error handling

waynetee committed:
Add delete

waynetee committed:
Add storage

waynetee committed:
Add dateTime

waynetee committed:
Merge branch 'branch-Level-7'

waynetee committed:
Merge branch 'branch-Level-8'

waynetee committed:
Add Ui

waynetee committed:
Add TaskList

waynetee committed:
Add Parser

waynetee committed:
Add package

waynetee committed:
Add tests

waynetee committed:
Add manifest

waynetee committed:
Fix access modifiers

waynetee committed:
Rename done to finishTask

waynetee committed:
Rename filePath to fileName

waynetee committed:
Add javadoc

waynetee committed:
Refactor Tasks

waynetee committed:
Fix test scripts

waynetee committed:
Add find

waynetee committed:
Merge branch 'branch-A-JavaDoc'

waynetee committed:
Merge branch 'branch-A-CodingStandard'

waynetee committed:
Merge branch 'branch-Level-9'

waynetee committed:
Refactor

waynetee committed:
Fix fileName

waynetee committed:
Merge branch 'add-gradle-support'

waynetee committed:
Fix gradle mainClassName

waynetee committed:
Add checkstyle config

waynetee committed:
Make checkstyle happy

waynetee committed:
Silence checkstyle

waynetee committed:
Remove emoji

waynetee committed:
Fix tests

waynetee committed:
Refactor Duke

waynetee committed:
Refactor Duke further

waynetee committed:
Add gui

waynetee committed:
Fix dialogContainer size

waynetee committed:
Fix daduke

waynetee committed:
Add javadoc

waynetee committed:
Make checkstyle happy

waynetee committed:
Add punctuation to javadoc

EltonGohJH reviewed (commented):
Hi Wayne,

LGTM. I legit cannot find any violation of other code quality guideline 😆. Your code is very easy to navigate and the logic is very clear.  Nice work and all the best  :grinning:!

MarcusTXK commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
```
Consider writing a descriptive header comment for Deadline class? I noticed that you may have missed adding header comments for some other classes as well, so perhaps it would be good to consistently write descriptive header comments for all public classes/methods.

MarcusTXK commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,32 @@
+    /**
+     * Shows output to user.
+     *
+     * @param lines lines to show user.
```
Consider using capital letters at the start of the sentence here for consistency with the rest of the Javadocs?

```suggestion
     * @param lines Lines to show user.
```

waynetee replied:
Nice catch! You have a sharp eye :)

MarcusTXK reviewed (commented):
Overall great code! I like how it is structured and it is easy to read and understand. 

LGTM, almost. Just a few really minor nitpicks to fix (I could not find any significant coding standard violation, nice job!).

waynetee committed:
Fix javadoc capitalisation

waynetee commented:
Thanks both for your comments!

waynetee committed:
Add assertions

There are some assumptions in the code of the Duke class

Let's add some assertions to validate them

waynetee committed:
Refactor code

Currently, the main method does not "make the happy path prominent"

Let's refactor it with guard clauses

waynetee committed:
Rename variable

Currently, the finishTask method has a badly named local variable

Let's refactor it to make the code more readable

waynetee committed:
Merge pull request #2 from waynetee/branch-A-Assertions

Add assertions

waynetee committed:
Merge branch 'master' into branch-A-CodeQuality

waynetee committed:
Merge pull request #1 from waynetee/branch-A-CodeQuality

Improve code quality

waynetee committed:
Add CI

waynetee committed:
Add aliases (#3)

waynetee committed:
Add Ui.png

waynetee committed:
Add auto flushing of output to disk

waynetee committed:
Add user guide

waynetee committed:
Set theme jekyll-theme-minimal

waynetee committed:
Set theme jekyll-theme-merlot

waynetee committed:
Set theme jekyll-theme-leap-day

waynetee committed:
Delete theme

waynetee committed:
Update user guide

waynetee committed:
Improve error messages

waynetee committed:
Fix font on macos

waynetee committed:
Update user guide
